scipy/stats/tests/test_stats.py

```diff
@@ -1469,7 +1469,7 @@ def test_regress_alternative(self):
 
         # slope is greater than zero, so "greater" p-value should be small
         res3 = stats.linregress(x, y, alternative="greater")
-        assert_allclose(res3.pvalue, res1.pvalue / 2)
+        assert_allclose(res3.pvalue, res1.pvalue / 2, atol=1e-16)
```

Comment:
The p-value is `2.072831043328079e-92`, so if there is an `atol` here, the test is not very sensitive.

This is sort of like the issue in the `ttest_rel` test above. We should have tested p-values that are a little less extreme.

Was this change necessary?
 (probably not now that we're implementing `_ttest_finish` with `stdtr`?)

Reply:
> Was this change necessary?
> (probably not now that we're implementing `_ttest_finish` with `stdtr`?)

No, this was only necessary when we were using `betainc` in `_ttest_finish`. With `betainc -> stdtr` change, we can revert this.